distribution/ddf-common/src/main/resources/bin/ddf.bat

```diff
@@ -3,42 +3,68 @@ SETLOCAL
 
 SET ARGS=%*
 SET DIRNAME=%~dp0%
-SET SOLR_PID="0"
+SET SOLR_PORT="8994"
+SET SOLR_MANAGED_INTERNALLY="true"
+SET SOLR_CLIENT=""
 
-PUSHD ..
+
+PUSHD %DIRNAME%\..
 SET DDF_HOME=%CD%
 POPD
 
-for /f "tokens=2 delims==" %%G in ('findstr /i "solr.http.port=" %DDF_HOME%\etc\system.properties') do (
-    SET SOLR_PORT=%%G
-)
 
 :RESTART
 REM Remove the restart file indicator so we can detect later if restart was requested
 IF EXIST "%DIRNAME%\restart.jvm" (
   DEL "%DIRNAME%\restart.jvm"
 )
 
-ECHO Starting Solr on port %SOLR_PORT%
-CALL %DDF_HOME%/solr/bin/solr.cmd start -p %SOLR_PORT%
-IF NOT ERRORLEVEL 0 (
-    ECHO WARNING! Solr start process returned non-zero error code, please check solr logs
+REM Get Solr port to run on
+for /f "tokens=2 delims==" %%G in ('findstr /i "^\w*solr.http.port=" %DDF_HOME%\etc\system.properties') do (
+    SET SOLR_PORT=%%G
+)
+
+REM Get Solr managed internally property
+for /f "tokens=2 delims==" %%G in ('findstr /i "^\w*solr.managed.internally=" %DDF_HOME%\etc\system.properties') do (
+    SET SOLR_MANAGED_INTERNALLY=%%G
+)
+
+REM Get Solr client property
+for /f "tokens=2 delims==" %%G in ('findstr /i "^\w*solr.client=" %DDF_HOME%\etc\system.properties') do (
+    SET SOLR_CLIENT=%%G
+)
+
+
+
+IF "%SOLR_MANAGED_INTERNALLY%" == "true" (
+    ECHO DEBUG REMOVE: Solr client is currently %SOLR_CLIENT%
+    IF NOT "%SOLR_CLIENT%" == "HttpSolrClient" (
+        ECHO ERROR! solr.managed.internally is set to true but the solr.client is not HttpSolrClient
+        ECHO Please set solr.managed.internally to false if you are not using the HttpSolrClient and
+        ECHO do not want solr to be managing the solr instance.
+        REM Exit code 83, for ascii code S, for Solr!
```

Comment:
OK. I kinda get 83. I think a return code of "1" might keep people from asking the question "why 83?"

Reply:
too many people use return code 1, rather have it designated a specific value in case anyone ever needs to react to this particular exit code.

Reply:
Line 44